apm-agent-attach/src/main/java/co/elastic/apm/attach/ElasticApmAttacher.java

```diff
@@ -151,14 +151,34 @@ public static void attach(String pid, Map<String, String> configuration, File ag
         File tempFile = createTempProperties(configuration, null);
         String agentArgs = tempFile == null ? null : TEMP_PROPERTIES_FILE_KEY + "=" + tempFile.getAbsolutePath();
 
-        ByteBuddyAgent.attach(agentJarFile, pid, agentArgs, ElasticAttachmentProvider.get());
+        attachWithFallback(agentJarFile, pid, agentArgs);
         if (tempFile != null) {
             if (!tempFile.delete()) {
                 tempFile.deleteOnExit();
             }
         }
     }
 
+    private static void attachWithFallback(File agentJarFile, String pid, String agentArgs) {
```

Comment:
do we have a test that shows the various attach failures?

Reply:
No we don't, mostly because that depends on the JVM its running on.

An integration test with a few docker images could help us test it, but I'm not sure it's really worth for now as we only had a single issue here so far.